demmt/nvrm_ioctl.h

```diff
@@ -516,6 +516,31 @@ struct nvrm_ioctl_unkd7 {
 };
 #define NVRM_IOCTL_UNKD7 _IOWR(NVRM_IOCTL_MAGIC, NVRM_IOCTL_ESC_BASE+15, struct nvrm_ioctl_unkd7)
 
+struct nvrm_ioctl_numa_info2 {
+	int32_t nid;
+	int32_t status;
+#define NVRM_IOCTL_NUMA_STATUS_DISABLED             0
+#define NVRM_IOCTL_NUMA_STATUS_OFFLINE              1
+#define NVRM_IOCTL_NUMA_STATUS_ONLINE_IN_PROGRESS   2
+#define NVRM_IOCTL_NUMA_STATUS_ONLINE               3
+#define NVRM_IOCTL_NUMA_STATUS_ONLINE_FAILED        4
+#define NVRM_IOCTL_NUMA_STATUS_OFFLINE_IN_PROGRESS  5
+#define NVRM_IOCTL_NUMA_STATUS_OFFLINE_FAILED       6
+	uint64_t memblock_size;
+	uint64_t numa_mem_addr;
+	uint64_t numa_mem_size;
+	struct {
+		uint64_t addresses[64];
+		uint32_t num_entries;
+	} blacklist_addresses;
+};
+#define NVRM_IOCTL_NUMA_INFO2 _IOWR(NVRM_IOCTL_MAGIC, NVRM_IOCTL_ESC_BASE+15, struct nvrm_ioctl_numa_info2)
```

Comment:
NVRM_IOCTL_UNKD7 above is probably the same ioctl, but without blacklist_addresses...

Reply:
I expected the same, but didn't have a setup with an older blob to verify this.

Reply:
I have only a few traces where UNKD7 shows up and in all of them demmt prints:
```
LOG: NVRM_IOCTL_UNKD7           pre,  fd: 7, unk00: 0x00000000, unk04: 0x00000000, unk08: 0x00000000, unk0c: 0x00000000, unk10: 0x00000000, unk14: 0x00000000, unk18: 0x00000000, unk1c: 0x00000000
LOG: NVRM_IOCTL_UNKD7           post, fd: 7, unk00: 0xffffffff, unk04: 0x00000000, unk08: 0x08000000, unk0c: 0x00000000, unk10: 0x00000000, unk14: 0x00000000, unk18: 0x00000000, unk1c: 0x00000000
```

so it's hard to say.

Reply:
390.67 and 390.77

Reply:
Added.